Source/WTF/wtf/RefPtr.h

```diff
@@ -254,25 +254,28 @@ struct IsSmartPtr<RefPtr<T, U, V>> {
 };
 
 template<typename ExpectedType, typename ArgType, typename PtrTraits, typename RefDerefTraits>
-inline bool is(RefPtr<ArgType, PtrTraits, RefDerefTraits>& source)
+inline bool is(const RefPtr<ArgType, PtrTraits, RefDerefTraits>& source)
 {
     return is<ExpectedType>(source.get());
 }
 
-template<typename ExpectedType, typename ArgType, typename PtrTraits, typename RefDerefTraits>
-inline bool is(const RefPtr<ArgType, PtrTraits, RefDerefTraits>& source)
+template<typename Target, typename Source, typename PtrTraits, typename RefDerefTraits>
+inline RefPtr<Target> downcast(RefPtr<Source, PtrTraits, RefDerefTraits>&& source)
 {
-    return is<ExpectedType>(source.get());
+    static_assert(!std::is_same_v<Source, Target>, "Unnecessary cast to same type");
+    static_assert(std::is_base_of_v<Source, Target>, "Should be a downcast");
+    ASSERT_WITH_SECURITY_IMPLICATION(!source || is<Target>(*source));
+    return static_reference_cast<Target>(WTFMove(source));
```

Comment:
Should this be `static_pointer_cast` instead of `static_reference_cast`?

Reply:
Yes, will fix.